### ISSUE TITLE
in POST v2/subscriptions in notification/attrs the subsc is created with attribute names equals

### ISSUE THREAD

**Issue by iariasleon** [telefonicaid/fiware-orion#2101]
in POST v2/subscriptions in notification/attrs with N attributes with the same name the subsc is created  but it should raise an 400 error. 
#### subscription request

```
POST http://localhost:1026/v2/subscriptions
Content-Type: application/json
Fiware-Service: test_notification_attrs
Fiware-ServicePath: /test
{ "subject": {"entities": [{"idPattern": ".*"}], "condition": { "attrs": ["temperature"] }}, "notification": {"http": { "url": "http://localhost:1234"}, "attrs": ["temperature","temperature","pressure","temperature","speed","temperature"]}, "expires": "2016-04-05T14:00:00.00Z", "throttling": 0 }
```
#### subscription response

```
 http code: 201
date: Wed, 27 Apr 2016 15:12:56 GMT
fiware-correlator: 853c0268-0c8a-11e6-b33c-005056a20feb
connection: Keep-Alive
content-length: 0
location: /v2/subscriptions/5721dc9aec637e4ec8fd2c7c
```
#### expected response

```
http code: 400
...
{"error":"BadRequest","description":"not defined yet"}
```
#### mongo docs

```
{ "_id" : ObjectId("5721dc9aec637e4ec8fd2c7c"), "expiration" : NumberLong(1459864800), "reference" : "http://localhost:1234", "throttling" : NumberLong(0), "servicePath" : "/test", "entities" : [ { "id" : ".*", "isPattern" : "true" } ], "attrs" : [ "temperature", "temperature", "pressure", "temperature", "speed", "temperature" ], "status" : "active", "conditions" : [ { "type" : "ONCHANGE", "value" : [ "temperature" ] } ], "expression" : { "q" : "", "geometry" : "", "coords" : "", "georel" : "" }, "format" : "normalized" }
```

**Comment by iariasleon** [comment]
the `notification` - `exceptAttrs` has the same behaviour:

#### request:
```
url: POST http://qa-orion-fe-02:1026/v2/subscriptions
headers:
    Content-Type: application/json
    Fiware-Service: test_notif_except_attrs_error
    Fiware-ServicePath: /test
payload: { "subject": {"entities": [{"idPattern": ".*"}], "condition": { "attrs": ["temperature"] }}, "notification": {"http": { "url": "http://localhost:1234"}, "exceptAttrs": ["temperature","temperature"]}, "expires": "2016-04-05T14:00:00.00Z" }
````
#### Response 
```
http code: 201 - Created
headers:
   date: Wed, 15 Jun 2016 15:59:51 GMT
   fiware-correlator: 31af1d2a-3312-11e6-a816-005056a20feb
   connection: Keep-Alive
   content-length: 0
   location: /v2/subscriptions/57617b77e54ad512e6b465b2
payload:
```

**Comment by iariasleon** [comment]
Re-tested in the CB version. It issue still fails and returns `201-Created`
```
  "version" : "1.2.0-next",
  "git_hash" : "d81dcf28e71d8d63dc0929ef7a6b73d7db47421b"
```

**Comment by kzangeli** [comment]
While it is true we could easily fix this issue, I can't say I see any reason to.

The broker only does what it is asked to do.

Very corner-case and to avoid this situation the broker would have to compare all attrs with each other. We'd make the broker less efficient just to fix something that is not very interesting.

IMO this issue should be closed.

**Comment by fgalan** [comment]
Yes, very corner case.

I wouldn't close the issue (yet) but it shouldn't be consisered a bug. Move to tech debt nice to have.

**Comment by Anjali-NEC** [comment]
Hi @fgalan sir,

I would like to know this issue is still valid or not, so that I will start working on it.

**Comment by fgalan** [comment]
Yes, the issue is still valid. It should be easy to solve: just checking for duplication at parsing stage (using std::map, for instance).

Thanks for your willingness to work on it! :)

**Comment by Anjali-NEC** [comment]
> Yes, the issue is still valid. It should be easy to solve: just checking for duplication at parsing stage (using std::map, for instance).
> 
> Thanks for your willingness to work on it! :)

I have started working on it. please assign this issue to me.

**Comment by fgalan** [comment]
Assigned. Thanks!

**Comment by Anjali-NEC** [comment]
@fgalan sir,

I have fixed this issue in PR #4025. Please review my PR, if it is ok then please merge it into master.
